src/sentry/workflow_engine/endpoints/validators/metric_alert_detector.py

```diff
@@ -38,3 +43,76 @@ def validate(self, attrs):
         if len(conditions) > 2:
             raise serializers.ValidationError("Too many conditions")
         return attrs
+
+    def update_data_conditions(self, instance, data_conditions):
+        """
+        Update the data condition if it already exists, create one if it does not
+        """
+        if instance.workflow_condition_group:
+            try:
+                data_condition_group = DataConditionGroup.objects.get(
+                    id=instance.workflow_condition_group.id
+                )
+            except DataConditionGroup.DoesNotExist:
+                raise serializers.ValidationError("DataConditionGroup not found, can't update")
+        # TODO make one if it doesn't exist and data is passed?
+
+        for data_condition in data_conditions:
+            try:
+                current_data_condition = DataCondition.objects.get(
+                    id=data_condition.get("id"), condition_group=data_condition_group
+                )
+            except DataCondition.DoesNotExist:
+                current_data_condition = None
+
+            if current_data_condition:
+                current_data_condition.update(**data_condition)
+                current_data_condition.save()
+            else:
+                DataCondition.objects.create(
+                    type=data_condition.get("type"),
+                    comparison=data_condition.get("comparison"),
+                    condition_result=data_condition.get("condition_result"),
+                    condition_group=data_condition_group,
+                )
+        return data_condition_group
+
+    def update_data_source(self, instance, data_source):
```

Comment:
can we add types here to make it a little more clear about the `data_source`? it's not clear if it's a dict that represents a DataSource's data, an instance of a `DataSource` (which is what i would assume based on the name), or if it's the new snuba query to become the data_source (based on the method, that's a bit more what i'd expect).